### ISSUE TITLE
Repeated grids/sections: make "Add Another" link clearer

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#3622]
Suggesting:

- make text clearer
- adding `role="button"` on link buttons

[+1 from customer](https://3.basecamp.com/3600924/buckets/1966559/messages/676074626#__recording_1125333064)

**Comment by ebruchez** [comment]
See also #2652, where @avernet went with `role="link"`, but I do think that `role="button"` is the better solution.

**Comment by avernet** [comment]
When to use [`role="link"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_link_role) vs. [`role="button"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role) isn't necessarily well defined. It seems we should use `role="link"` if a click doesn't change the focus. And we shouldn't `role="button"` if the action can't be invoked by pressing space.

Then what if the action doesn't change the focus *and* can't be invoked with space? I imagine that we should then either change the focus to something reasonable and use `role="link"`, or make it so it can be invoked with space and  use `role="button"`.

**Comment by ebruchez** [comment]
The screen reader announces the role as either link or button based on the role. From the (blind, especially) user's point of view, in the case of our link buttons, they do behave like buttons, and you don't care that they may look like a link: it's only a styling thing. What matters is the label and that activating the button will do the same thing that a regular button will do.

VoiceOver, by the way, tells you right there how to activate the button: there is a special key combination for that (command-option-space). Space or enter doesn't work.

Whether we should be able to activate a link button with space remains to be seen.

I am still more than 50% in favor of `role="button"`.

**Comment by avernet** [comment]
> Whether we should be able to activate a link button with space remains to be seen.

The [Keyboard and focus](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role#Keyboard_and_focus) section provides an answer to that question.
